src/tx-format/transaction.md

```diff
@@ -159,3 +160,36 @@ Transaction is invalid if:
 
 - `txPointer` is zero or doesn't match the block.
 - `outputContract.inputIndex` is not zero
+
+## `TransactionUpgrade`
+
+| name             | type                                   | description                    |
+|------------------|----------------------------------------|--------------------------------|
+| `upgradePurpose` | [UpgradePurpose](./upgrade_purpose.md) | The purpose of the upgrade.    |
+| `policyTypes`    | `uint32`                               | Bitfield of used policy types. |
+| `inputsCount`    | `uint16`                               | Number of inputs.              |
+| `outputsCount`   | `uint16`                               | Number of outputs.             |
+| `witnessesCount` | `uint16`                               | Number of witnesses.           |
+| `policies`       | [Policy](./policy.md)`[]`              | List of policies.              |
+| `inputs`         | [Input](./input.md)`[]`                | List of inputs.                |
+| `outputs`        | [Output](./output.md)`[]`              | List of outputs.               |
+| `witnesses`      | [Witness](./witness.md)`[]`            | List of witnesses.             |
+
+Given helper `max_gas()` returns the maximum gas that the transaction can use.
+Given helper `count_ones()` that returns the number of ones in the binary representation of a field.
+Given helper `count_variants()` that returns the number of variants in an enum.
+Given helper `sum_variants()` that sums all variants of an enum.
```

Comment:
We're repeating these quite a bit. Maybe we should have a centralized list of these? Although I think these might be almost evident enough tha they don't need explanations.